### PR TITLE
Add pretrained Wide ResNet

### PR DESCRIPTION
**Pull request by szagoruyko**
I trained WRN-50-2 and WRN-101-2 with master torchvision, which now allows making WRN models with simple width_per_group argument. I did not use the standard training procedure for ResNet though, here are the differences:
 * SGD with cosine learning rate and warm restarts for 256 epochs (~0.2% to top1 accuracy)
 * FP16 training with batchnorm in FP32 with apex O2

so the checkpoints are in torch.float16 to save space.

| model | top1, top5 error |
|:---|:---:|
| WRN-50-2 | 21.49, 5.91 |
| WRN-101-2 | 21.16, 5.72 |

idk do we want these in torchvision? I could put them in wide-residual-networks instead.

### CODE REVIEW

File: torchvision/models/resnet.py [hunk 1]
```diff
@@ -294,3 +296,15 @@ def resnext101_32x8d(pretrained=False, progress=True, **kwargs):
     kwargs['width_per_group'] = 8
     return _resnet('resnext101_32x8d', Bottleneck, [3, 4, 23, 3],
                    pretrained, progress, **kwargs)
+
+
+def wide_resnet50_2(pretrained=False, progress=True, **kwargs):
+    kwargs['width_per_group'] = 64 * 2
```
Comment (fmassa) [inline torchvision/models/resnet.py]: Can you add some documentation for those functions? They can be very similar to the ones from ResNet

**Comment by szagoruyko** [comment]
Thanks for looking at it @fmassa , I added:
 * docstrings for wide_resnet50_2 and wide_resnet101_2 functions (with explanation of diff vs ResNet)
 * docs at docs/source/models.rst
 * wide_resnet50_2 and wide_resnet101_2 are now properly imported

**Comment by fmassa** [comment]
Sorry for the delay in replying.

I've uploaded the pre-trained weights to
```
https://download.pytorch.org/models/wide_resnet50_2-95faca4d.pth
https://download.pytorch.org/models/wide_resnet101_2-32ee1156.pth
```

Can you update the URLs and fix the conflicts? Then it's good to go!

**Comment by szagoruyko** [comment]
@fmassa done, updated links and rebased on master

**Comment by fmassa** [comment]
Thanks Sergey!
